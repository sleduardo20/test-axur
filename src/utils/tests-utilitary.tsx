import { ThemeProvider } from 'styled-components';
import { render, RenderOptions } from '@testing-library/react';

import theme from 'styles/theme';

import { ReactElement } from 'react';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const renderCustum = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';

export { renderCustum as render };
