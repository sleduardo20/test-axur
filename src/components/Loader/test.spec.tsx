import { render, screen } from '../../utils/tests-utilitary';

import { Loader } from '.';

describe('<Loader />', () => {
  it('should render Loader with icon', () => {
    render(<Loader />);

    expect(screen.getByLabelText(/loading/i)).toMatchSnapshot();
  });
});
