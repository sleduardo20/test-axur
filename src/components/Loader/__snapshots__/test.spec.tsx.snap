// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render Loader with icon 1`] = `
.c0 {
  z-index: 50;
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}

.c0 div {
  position: absolute;
  top: 33px;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background: #ff5824;
  -webkit-animation-timing-function: cubic-bezier(0,1,1,0);
  animation-timing-function: cubic-bezier(0,1,1,0);
}

.c0 div:nth-child(1) {
  left: 8px;
  -webkit-animation: lds-ellipsis1 0.6s infinite;
  animation: lds-ellipsis1 0.6s infinite;
}

.c0 div:nth-child(2) {
  left: 8px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c0 div:nth-child(3) {
  left: 32px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c0 div:nth-child(4) {
  left: 56px;
  -webkit-animation: lds-ellipsis3 0.6s infinite;
  animation: lds-ellipsis3 0.6s infinite;
}

<div
  aria-label="loading"
  class="c0"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
